src/app/game/game-intermediate-result/game-intermediate-result-header/confirm-exit-dialog/confirm-exit-dialog.component.ts

```diff
@@ -0,0 +1,46 @@
+import { TranslatePipe } from '@/app/core/translation.pipe';
+import { GameStateService } from '@/app/game/services/game-state-service';
+import { routes } from '@/app/shared/models/routes';
+import { Router } from '@angular/router';
+import { MultiplayerService } from '@/app/game/services/multiplayer.service';
+import { Component, inject, ViewEncapsulation } from '@angular/core';
+import { MatButton } from '@angular/material/button';
+import { MatDialogActions, MatDialogContent, MatDialogRef } from '@angular/material/dialog';
+import { MatIcon } from '@angular/material/icon';
+
+@Component({
+  selector: 'app-confirm-exit-dialog',
+  templateUrl: './confirm-exit-dialog.component.html',
+  styleUrls: ['./confirm-exit-dialog.component.scss'],
+  encapsulation: ViewEncapsulation.None,
+  standalone: true,
+  imports: [MatDialogActions, MatDialogContent, MatButton, MatIcon, TranslatePipe],
+})
+export class ConfirmExitDialogComponent {
+  constructor(
+    private router: Router,
+    private gameStateService: GameStateService,
+    private multiplayerService: MultiplayerService
+  ) {}
+
+  readonly dialogRef = inject(MatDialogRef<ConfirmExitDialogComponent>);

```

Comment:
I like how you used inject here. Clever solution by a clever man 🤤

Reply:
Cannot take credz for this. Inspired by the Angular docs!